main.py

```diff
@@ -204,7 +204,26 @@ def to_groups(item_list, config):
     else:
         for item_set in sets:
             groups.append(sets[item_set])
-    groups.append(uniques)
+    
+    if config["SETTINGS"]["UnifyUniques"] == '1':
```

Comment:
This code is overly long and complicated. If you want a UnifyUniques option, then just use the same method I used for sets. Turn uniques into a dictionary instead of a list, and add each item to uniques[type]. Then, add another helper function called to_supergroup (or something like that) which unifies a dictionary, i.e.
def to_supergroup(group_dict)
   supergroup = []
   for subgroup in group_dict:
      supergroup.extend(sets[item_set])
   return supergroup

then you can just have a similar structure for sets and uniques: 

if config["SETTINGS"]["UnifySets"] == '1':
   groups.append(to_supergroup(sets)
else:
   for item_set in sets:
   groups.append(sets[item_set])

and the same for uniques. It would be better to refactor these into their own functions, too.
While I like your idea, I also wonder if it should really be a setting. Perhaps it should be made the only option, since I can't really imagine the benefit of not dividing uniques into types.